### PR TITLE
Account Settings page for v0.7

### PR DESCRIPTION
**Pull request by NickFranciscoOSU**
Added an account edit modal that allows the user to edit their email in Firebase Auth or other information in the database. I will implement password change and clean up the page for next revision. I also have a temporary fix for a lint error on the accountSettings page that will be resolved in the next revision.

### CODE REVIEW

**Comment by cdgco** [comment]
I'm getting the following error when attempting to update the email address `Uncaught (in promise) FirebaseError: Firebase: Error (auth/requires-recent-login).` I believe updating an email address is considered a sensitive action so the user has to login again in order to update it. See https://firebase.google.com/docs/auth/web/manage-users#re-authenticate_a_user.

Also, let me know what you think about this, but I was thinking of using the `email` property in the `shelters` collection as like a separate public contact email address that might be different from the auth email address. I'm fine using just one address for both, but I had edited the one in the database on my account to be a different address and noticed that the settings page shows the address from firebase auth, but when I open the form, it shows the address from the database.

**Comment by NickFranciscoOSU** [comment]
I made it so the user is redirected back to the login page if their authentication token is expired. I would like to do something different in the future instead of redirecting, like a pop up that asks them to re-enter their password. I will make this change next week when I implement the password change feature which will require a recent login as well. 

I am ok having an email address in the database to represent the public email address. I will make it so there are separate place to update their public email or login email in a future update.

File: src/components/Cards/EditAccountModal.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        if (curAccountBody.email !== auth.currentUser.email) {
+
+            try {
+                await updateEmail(auth.currentUser, curAccountBody.email);
+            }
+            // Redirect back to login if authentication session expired
+            catch (e) {
+                if(e) {
+                    signOut(auth).then(() => {
+                        navigate("/login")
```
Comment (cdgco) [inline src/components/Cards/EditAccountModal.js:44]: I think that should go to `sign-in` instead of `login`

**Review by cdgco — CHANGES_REQUESTED** [review]

File: src/components/Cards/EditAccountModal.js (hunk 1 above)
Comment (NickFranciscoOSU) [reply src/components/Cards/EditAccountModal.js]: Nice catch, I'm not sure how it worked when I was testing it.

**Review by cdgco — APPROVED** [review]

**Pull request merged**